### DISCUSSION TITLE
[Docker] ALPINE_DATABASE_PASSWORD from the secret

### DISCUSSION THREAD

**Discussion by gray380** [DependencyTrack/dependency-track#1987] [Q&A]
Hello,

Is there a possibility to set `ALPINE_DATABASE_PASSWORD` from the docker secret?
I mean in the `docker-compose.yml`.

best regards
Serhiy.

**Comment by nscuro** [comment]
Please don't create issues for questions, see https://github.com/DependencyTrack/dependency-track/blob/master/CONTRIBUTING.md#asking-questions

**Comment by nscuro** [comment]
@gray380 I just checked, and while it's not really documented, you can use the `ALPINE_DATABASE_PASSWORD_FILE` environment variable to specify a file from where the password shall be loaded.

If your Docker secret has the name `database_password`, then you'd set `ALPINE_DATABASE_PASSWORD_FILE` to `/run/secrets/database_password`.

The same is possible for other configuration options where passwords are provided, e.g.:

```
ALPINE_LDAP_BIND_PASSWORD_FILE
ALPINE_HTTP_PROXY_PASSWORD_FILE
```

**Reply by mendorf** [reply]
I also spent some time to find out how to make this work, probably this will help someone still searching on how to get the secret consumed by Dependency Track.

It is not clear from the documentation that the *_FILE property keys even exist. However, even if there seems to be some [code](https://github.com/stevespringett/Alpine/blob/b0f9b6a83feaf5b6f5e82410eeed2ebbb68a2509/alpine-common/src/main/java/alpine/Config.java#L424) the keys do not work at all. 

Let alone ALPINE_APPLICATION_PROPERTIES. One needs to add it to the java runtime, which is only mentioned vaguely inside [documentation](https://github.com/DependencyTrack/dependency-track/blob/8a5aee4451fa7209476b28a182f122228c9e9514/docs/_docs/getting-started/configuration.md?plain=1#L33). 

This approach renders a _FILE property useless, because you basically are now able to directly include the password within the application.properties, if you do not care to have the remaining configuration details readable for others.
There is a similar issue #832 already open since 2020.

Hope this helps anyone to save time to figure out.

**Reply by nscuro** [reply]
@mendorf Just looked into it. `ALPINE_DATABASE_PASSWORD_FILE` not being used is a regression in Alpine. Raised a PR to resolve this: https://github.com/stevespringett/Alpine/pull/557
